table.cc

```diff
@@ -1199,16 +1199,18 @@ table::on_compaction_completion(const std::vector<sstables::shared_sstable>& new
 
     rebuild_statistics();
 
-    // This is done in the background, so we can consider this compaction completed.
-    (void)seastar::with_gate(_sstable_deletion_gate, [this, sstables_to_remove] {
-       return with_semaphore(_sstable_deletion_sem, 1, [this, sstables_to_remove = std::move(sstables_to_remove)] {
-        return sstables::delete_atomically(sstables_to_remove).then_wrapped([this, sstables_to_remove] (future<> f) {
-            std::exception_ptr eptr;
-            try {
-                f.get();
-            } catch(...) {
-                eptr = std::current_exception();
-            }
+    auto f = seastar::with_gate(_sstable_deletion_gate, [this, sstables_to_remove] {
+       return with_semaphore(_sstable_deletion_sem, 1, [sstables_to_remove = std::move(sstables_to_remove)] {
+           return sstables::delete_atomically(std::move(sstables_to_remove));
+       });
+    });
+
```

Comment:
This was originally done in the background because the atomic_deletion_manager could impose unbounded latency, but now that it is gone, I don't think it's a problem.